### PR TITLE
Tweaks for synapse density support

### PR DESCRIPTION
**Pull request by tribal-tec**
- Only load volume once
- Print datasource in volume statistics (separated from statistics with 'i' key)
- Only print data units if provided
- ICC build fixes
- Use less pipeline threads to avoid multiple sampling of the same bricks
- Silence warning on startup if no transfer function was provided

### CODE REVIEW

File: livre/core/data/VolumeInformation.h [hunk 1]
```diff
@@ -109,7 +109,8 @@ struct VolumeInformation
       */
     Vector2ui frameRange;
 
-
+    /** Optional description filled by datasource which describes the data. */
```
Comment (eile) [inline livre/core/data/VolumeInformation.h]: Optional description for end users?

Comment (tribal-tec) [inline livre/core/data/VolumeInformation.h]: Done

**Comment by tribal-tec** [comment]
Any chance for a +1 after 1 week?

File: livre/eq/Channel.cpp [hunk 2]
```diff
@@ -355,22 +356,58 @@ struct Channel::Impl
...
+           << "Total resolution " << info.voxels << " depth "
+           << info.rootNode.getDepth() << std::endl
+           << "Block resolution " << info.maximumBlockSize << std::endl;
+
+        std::string unit;
+        if( info.meterToDataUnitRatio == 1e6 )
+            unit = "um";
+        else if( info.meterToDataUnitRatio == 1e3 )
+            unit = "mm";
+        else if( info.meterToDataUnitRatio == 1.0f )
```
Comment (eile) [inline livre/eq/Channel.cpp]: Floating comparison, mixed 'f' usage.

File: livre/eq/Channel.cpp [hunk 3]
```diff
@@ -355,22 +356,58 @@ struct Channel::Impl
...
+            unit = "m";
+        const Vector3f resolution = info.resolution;
```
Comment (eile) [inline livre/eq/Channel.cpp]: const &

File: livre/eq/Node.cpp [hunk 4]
```diff
@@ -130,17 +130,19 @@ bool Node::configInit( const eq::uint128_t& initId )
...
     livre::Client* client = static_cast<livre::Client*>( getClient( ).get());
     client->setIdleFunction( std::bind( &Impl::updateAndSendFrameRange,
                                         _impl.get()));
 
-    if( !isApplicationNode( ))
-    {
-        Config *config = static_cast< Config *>( getConfig() );
+    Config *config = static_cast< Config *>( getConfig() );
+    if( isApplicationNode( ))
+        config->onNewVolume( _impl->_dataSource->getVolumeInfo( ));
```
Comment (eile) [inline livre/eq/Node.cpp]: needs to be eq::Config->sendEvent() and handler

**Comment by eile** [comment]
+1 otherwise

**Comment by tribal-tec** [comment]
Updated

File: livre/eq/Channel.cpp [hunk 5]
```diff
@@ -355,22 +356,59 @@ struct Channel::Impl
...
+           << info.rootNode.getDepth() << std::endl
+           << "Block resolution " << info.maximumBlockSize << std::endl;
 
-        glMatrixMode( GL_PROJECTION );
-        glLoadIdentity();
-        _channel->applyScreenFrustum();
-        glMatrixMode( GL_MODELVIEW );
+        std::string unit;
+        const float epsilon = std::numeric_limits< float >::epsilon();
+        if( (info.meterToDataUnitRatio - 1e6f) <= epsilon )
```
Comment (eile) [inline livre/eq/Channel.cpp]: if( std::abs( ... ) <= epsilon )

File: livre/eq/coSerialization.h [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+template<> inline void byteswap( livre::DataType& value )
+{
+    byteswap( reinterpret_cast< uint32_t& >( value ));
+}
+
+}
+
+#endif // _coSerialization_h_
```
Comment (eile) [inline livre/eq/coSerialization.h]: Since this only depends on Collage, why is it not in core with the struct?

**Pull request merged**